backend/src/test/java/com/google/sps/FindMatchQueryTest.java

```diff
@@ -0,0 +1,151 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import static com.google.common.truth.Truth.assertThat;
+import com.google.sps.data.Match;
+import com.google.sps.data.Participant;
+import com.google.sps.data.Time;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.Calendar;
+import java.util.Objects;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import static org.mockito.Mockito.*;
+
+/** */
+@RunWith(JUnit4.class)
+public final class FindMatchQueryTest {
+
+  // Some people that we can use in our tests
+  private static final String PERSON_A = "Person A";
+  private static final String PERSON_B = "Person B";
+  private static final String PERSON_C = "Person C";
+  private static final String PERSON_D = "Person D";
+
+  // Default parameters unused in query
+  private static final long ID_DEFAULT = 0;
+  private static final String TIMEZONE_DEFAULT = "";
+  private static final long TIMESTAMP_DEFAULT = 0;
+
+  // Some times available until on 6/25/2020
+  private static final long TIME_0250PM = Time.getTimeMillis(14, 50);
+  private static final long TIME_0330PM = Time.getTimeMillis(15, 30);
+  private static final long TIME_0400PM = Time.getTimeMillis(16, 0);
+  private static final long TIME_0600PM = Time.getTimeMillis(18, 0);
+
+  private static final int DURATION_15_MINUTES = 15;
+  private static final int DURATION_30_MINUTES = 30;
+  private static final int DURATION_45_MINUTES = 45;
+  private static final int DURATION_60_MINUTES = 60;
+
+  private FindMatchQuery query;
+  private Time time;
+
+  @Before
+  public void setUp() {
+    // Set "current" date to  1/1/2020 2:00pm GMT
+    Calendar c = Calendar.getInstance();
+    c.set(2020, 0, 1, 14, 0, 0);
+    c.set(Calendar.MILLISECOND, 0);
+    Date date = c.getTime();
+
+    query = new FindMatchQuery(date);
+    time = new Time(date);
+  }
+
+  @Test
+  public void compatibleTimeAndDuration() {
+    /* Two participants who are compatible in available time AND duration */
+    Participant participantA = new Participant(ID_DEFAULT, PERSON_A, TIME_0400PM, TIMEZONE_DEFAULT, DURATION_30_MINUTES, TIMESTAMP_DEFAULT);
+    Participant participantB = new Participant(ID_DEFAULT, PERSON_B, TIME_0600PM, TIMEZONE_DEFAULT, DURATION_15_MINUTES, TIMESTAMP_DEFAULT);
+    
+    Match actual = query.findMatchQuery(Arrays.asList(participantA, participantB));
```

Comment:
here and below:

internally we have a convention to use ImmutableList from Google collections when passing lists in tests. Not only is the API slightly shorter, its use also helps catch bugs where methods unexpectedly mutate lists.

Reply:
Great suggestion, (I think) I made this change? I just did query.findMatchQuery(ImmutableList.copyOf(Arrays.asList(participantA, participantB)));